docs/src/main/asciidoc/security-authorize-web-endpoints-reference.adoc

```diff
@@ -46,6 +46,37 @@ quarkus.http.auth.policy.role-policy1.roles-allowed=user,admin
 ----
 <1> This defines a role-based policy that allows users with the `user` and `admin` roles.
 
+Sometimes it might be useful to register your own named policy. You can get it done by creating application scoped CDI
+bean that implements the `io.quarkus.vertx.http.runtime.security.HttpSecurityPolicy` interface like in the example below:
+
+[source,java]
+----
+import jakarta.enterprise.context.ApplicationScoped;
+
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.vertx.http.runtime.security.HttpSecurityPolicy;
+import io.smallrye.mutiny.Uni;
+import io.vertx.ext.web.RoutingContext;
+
+@ApplicationScoped
+public class CustomNamedHttpSecPolicy implements HttpSecurityPolicy {
+    @Override
+    public Uni<CheckResult> checkPermission(RoutingContext request, Uni<SecurityIdentity> identity,
+            AuthorizationRequestContext requestContext) {
+        if (customRequestAuthorization(request)) {
+            return Uni.createFrom().item(CheckResult.PERMIT);
+        }
+        return Uni.createFrom().item(CheckResult.DENY);
+    }
+
+    @Override
+    public String name() {
+        return "custom"; <1>
+    }
+}
+----
+<1> Named HTTP Security policy will only be applied to requests matched by the `application.properties` path matching rules.
```

Comment:
Please also clarify that users don't have to name them, if they don't then they apply to all authorization checks

Reply:
Right, I was thinking about that as AFAICT it is nowhere documented. But laziness got worse of me. Will do.

Reply:
I added a TIP.